pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/TableDataManagerProvider.java

```diff
@@ -51,13 +52,17 @@ public static TableDataManager getTableDataManager(@Nonnull TableDataManagerConf
       @Nonnull String instanceId, @Nonnull ZkHelixPropertyStore<ZNRecord> propertyStore,
       @Nonnull ServerMetrics serverMetrics, @Nonnull HelixManager helixManager) {
     TableDataManager tableDataManager;
-    switch (TableType.valueOf(tableDataManagerConfig.getTableDataManagerType())) {
+    switch (TableDataManagerType.valueOf(tableDataManagerConfig.getTableDataManagerType())) {
       case OFFLINE:
         tableDataManager = new OfflineTableDataManager();
         break;
       case REALTIME:
         tableDataManager = new RealtimeTableDataManager(_segmentBuildSemaphore);
         break;
+      case DIMENSION:
+        // TODO: Create a DimTableDataManager here when available
```

Comment:
do we need `DimTableDataManager`? or enhance `OfflineTableDataManager `

Reply:
`DimTableManager` will have a lot of specific themes related to preloading the table and also storing the dim table static map, thus it will muddy the `OfflineTableDataManager` unnecessarily. Thus, we are thinking of extending the `OfflineTableDataManager` in the new manager class to implement the extra functionality. We can discuss this in following PRs as well.